core/src/style/style.cpp

```diff
@@ -69,10 +79,59 @@ void Style::setLightingType(LightingType _type){
 
 }
 
-void Style::addLayer(const std::pair<std::string, StyleParamMap>&& _layer) {
+void Style::addLayer(SceneLayer* _layer) {
+
+    m_layers.push_back(_layer);
+
+}
+
+void Style::applyLayerFiltering(const Feature& _feature, const Context& _ctx, long long& _uniqueID,
+                                   StyleParamMap& _styleParamMapMix, SceneLayer* _uberLayer) const {
+
+    std::vector<SceneLayer*> sLayers;
+    sLayers.reserve(_uberLayer->getSublayers().size() + 1);
+    sLayers.push_back(_uberLayer);
+
+    auto sLayerItr = sLayers.begin();
+
+    //A BFS traversal of the SceneLayer graph
+    while (sLayerItr != sLayers.end()) {
+
+        if ( (*sLayerItr)->getFilter()->eval(_feature, _ctx)) { // filter matches
+
+            _uniqueID += (1 << (*sLayerItr)->getID());
```

Comment:
There's not really a concept of order in which styles match. The order that matters is to ensure that properties from "deeper" styles take precedence over shallower ones. (There is ambiguity when multiple styles at the same depth level modify the same properties.) 

@blair1618 is right that we can't limit the # of styles prematurely though. Is it expensive to use a string key? Probably. Or maybe some kind of hashing function.